### PR TITLE
Refactor application

### PR DESCRIPTION
**Pull request by alaindet**
Proposed refactoring of the application

- Rescaffold according to https://github.com/golang-standards/project-layout
- Create cmd/* folder for executables
- Create /internal folder for internal packages
- Rename dabatase/connection.go to database/database.go according to best practices
- Use Singleton pattern for database connection pool
- Add .sql initialization file in docker-compose.yaml
- Update README.md
- Remove unused temporary variables in titles/controller.go

### CODE REVIEW

**Comment by alaindet** [comment]
- Add alaindet as second author

File: cmd/going-live/titles/controller.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package titles
```
Comment (advance467) [inline cmd/going-live/titles/controller.go:1]: Please move this back into "features".

**Review by advance467 — CHANGES_REQUESTED** [review]

File: docker-compose.yaml [hunk 2]
```diff
@@ -17,20 +17,26 @@ services:
       - database
 
   database:
-    image: 'postgres:15.1'
+    image: 'postgres:15.1-alpine'
```
[the remark this answers is not in the document]
Comment (alaindet) [reply docker-compose.yaml:20]: Nope LOL

File: cmd/going-live/titles/controller.go (hunk 1 above)
Comment (alaindet) [reply cmd/going-live/titles/controller.go:1]: ok

File: README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+docker-compose up --build
+```
+
+Try
+```console
+curl localhost:3001/titles | jq
+curl localhost:3001/titles/1 | jq
+```
+
+## Author
```
[the remark this answers is not in the document]
Comment (alaindet) [reply README.md:28]: ok

**Comment by alaindet** [comment]
Changes done

**Comment by alaindet** [comment]
Changes addressed

**Review by advance467 — APPROVED** [review]

**Comment by advance467** [comment]
Thanks!

**Pull request merged**